starlite/middleware/session/file_backend.py

```diff
@@ -0,0 +1,85 @@
+
+import anyio
+import orjson
+
+from starlite.middleware.session.base import ServerSideBackend, ServerSideSessionConfig
+
+FileStorageMetadataWrapper = Tuple[str, str]
+
+
+class FileBackend(ServerSideBackend["FileBackendConfig"]):
+    def __init__(self, config: "FileBackendConfig") -> None:
+        """Session backend to store data in files."""
+        super().__init__(config=config)
+        self.path = anyio.Path(config.storage_path)
+
+    def _id_to_storage_path(self, session_id: str) -> anyio.Path:
+        return self.path / session_id
+
+    @staticmethod
+    async def _load_from_path(path: anyio.Path) -> FileStorageMetadataWrapper:
+        wrapped_data: FileStorageMetadataWrapper = orjson.loads(await path.read_bytes())
+        return wrapped_data
+
+    @staticmethod
+    def _is_expired(wrapped_data: FileStorageMetadataWrapper) -> bool:
+        return datetime.fromisoformat(wrapped_data[0]) > datetime.utcnow().replace(tzinfo=None)
+
+    async def get(self, session_id: str) -> Optional[bytes]:
+        """Load data associate with `session_id` from a file.
+
+        If no such file exists, or the session has expired, return
+        `None`. If the session is expired, delete the file
+        """
+
+        path = self._id_to_storage_path(session_id)
+        if await path.exists():
+            wrapped_data = await self._load_from_path(path)
+            if self._is_expired(wrapped_data):
+                return wrapped_data[1].encode()
+            await path.unlink()
+        return None
+
+    async def set(self, session_id: str, data: bytes) -> None:
+        """Store `data` alongside metadata under the `session_id`, using the ID
+        as a filename.
+
+        If a file already exists for `session_id`, replace it with
+        `data` and reset its expiry time
+        """
+        await self.path.mkdir(exist_ok=True)
+        path = self._id_to_storage_path(session_id)
+        wrapped_data: FileStorageMetadataWrapper = (
+            (datetime.utcnow().replace(tzinfo=None) + timedelta(seconds=self.config.max_age)).isoformat(),
+            data.decode(),
+        )
+        await path.write_bytes(orjson.dumps(wrapped_data))
+
+    async def delete(self, session_id: str) -> None:
+        """Delete the file associated with `session_id`.
+
+        Fails silently if no such file exists
+        """
+        path = self._id_to_storage_path(session_id)
+        await path.unlink(missing_ok=True)
+
+    async def delete_all(self) -> None:
+        """Delete all files in the storage path."""
+        async for file in self.path.iterdir():
+            await file.unlink(missing_ok=True)
+
+    async def delete_expired(self) -> None:
+        """Delete expired session files."""
+        async for file in self.path.iterdir():
+            wrapper = await self._load_from_path(file)
+            if self._is_expired(wrapper):
+                await file.unlink()
+
+
+class FileBackendConfig(ServerSideSessionConfig):
```

Comment:
should we allow users to customize the filename for sessions ? might be over engineering at this point.

Reply:
Eh, why not? Added.